frontend/src/pages/ProcessDataPage/ProcessedDataPage.tsx

```diff
@@ -8,84 +8,259 @@ import {
   Button,
   Alert,
   Snackbar,
-  List,
   Container,
 } from "@mui/material";
-import { WatchType } from "shared/api";
+import { DataType, PredictionType, RawFileData, WatchType } from "shared/api";
 import useListUploadedFiles from "shared/hooks/useListUploadedFiles";
-
+import usePredictedFile from "shared/hooks/usePredictFile";
+// import useDeleteFile from "shared/hooks/useDeleteFile";
+// import { saveAs } from "file-saver";
+import { FileWithPath } from "react-dropzone";
 import styles from "./ProcessedDataPage.module.css";
 
 const ProcessedDataPage = function () {
   const rollbarConfig = {
     accessToken: "dbfced96b5df42d295242681f0560764",
-    environment: "production",
+    environment: "dev",
   };
   const rollbar = new Rollbar(rollbarConfig);
   rollbar.debug("Reached Processed Data page");
 
-  const { uploadedFiles } = useListUploadedFiles(WatchType.APPLE_WATCH);
-  console.log(uploadedFiles);
+  const [files, setFiles] = useState<any[]>([]);
+  const [currentFile, setCurrentFile] = useState<any>();
+  const [selectedModel, setSelectedModel] = useState<PredictionType>(
+    PredictionType.SVM,
+  );
+
+  const { handlePredict } = usePredictedFile();
+  // const { handleDelete } = useDeleteFile();
+
+  // the list of radial selectors for the file list
+  let renders: any;
+
+  /**
+   * This function will get the users processed files from the database
+   * PRE-Conditions: User is logged in and authenticated
+   * POST-Conditions: Files will be updated to be the users list of files
+   *                  The list of displayed files will be updated to show this
+   */
+  const getUploadedFiles = () => {
+    if (files.length === 0) {
+      try {
+        const { uploadedFiles } = useListUploadedFiles(WatchType.APPLE_WATCH);
```

Comment:
These hooks should only be called once and should live at the top of the file.

here is an alternative that would work just as well:

Delete this function and the `const [files, setFiles] = useState<any[]>([]);` because useListUploadedFiles essentially does the same thing that `const [files, setFiles] = useState<any[]>([]) `this is doing
You can do something like this instead which is  a much better equivalent

```
const files =
    uploadedFiles.length !== 0
      ? uploadedFiles.map((file: RawFileData) => ({
          ...file,
          watch: DataType.APPLE_WATCH,
        }))
      : [];

```